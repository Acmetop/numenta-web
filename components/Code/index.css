/**
 * Numenta shared web components
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2016 Numenta <http://numenta.com>
 */

.code {
  composes: bg-light-gray from 'tachyons';
  composes: db from 'tachyons';
  composes: f6 from 'tachyons';
  composes: lh-copy from 'tachyons';
  composes: mb3 from 'tachyons';
  composes: nowrap from 'tachyons';
  composes: overflow-auto from 'tachyons';
  composes: ph3 from 'tachyons';
  composes: pv2 from 'tachyons';
}
