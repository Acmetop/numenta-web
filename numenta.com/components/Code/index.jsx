import React from 'react'

import styles from './style.css'


const Code = ({children}) => (
  <code className={styles.code}>
    {children}
  </code>
)

Code.propTypes = {
  children: React.PropTypes.any.isRequired,
}

export default Code
