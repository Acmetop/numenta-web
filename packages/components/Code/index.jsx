// Numenta shared web components
// MIT License (see LICENSE.txt)
// Copyright © 2005—2016 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


/**
 *
 */
const Code = ({children}) => (
  <code className={styles.code}>
    {children}
  </code>
)

Code.propTypes = {
  children: React.PropTypes.node.isRequired,
}

export default Code
