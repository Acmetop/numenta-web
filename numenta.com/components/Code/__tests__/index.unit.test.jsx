import React from 'react'
import renderer from 'react-test-renderer'

import Code from '../../Code'


describe('Code React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Code>const KEY = 'id_123'</Code>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
