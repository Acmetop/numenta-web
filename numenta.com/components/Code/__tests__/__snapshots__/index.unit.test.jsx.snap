exports[`Code React component Renders correctly 1`] = `
<code
  className="code">
  const KEY = \'id_123\'
</code>
`;
